### ISSUE TITLE
pkg is deprecated

### ISSUE THREAD

**Issue by matteo-cristino** [ForkbombEu/ncr#384]
pkg has been deprecated 10 months ago, maybe it is time to move to something new and mantained like https://nodejs.org/api/single-executable-applications.html

**Comment by puria** [comment]
I know, we are the only one using that in the WORLD! But, I do prefer bun binary, better than nodes SEA, they are pretty new, and bho... I do prefer the `bun` one.

Anyhow, very low priority on this <3

**Comment by matteo-cristino** [comment]
I did some tests, with bun since `@slangroom/rdf` breaks rollup+pkg, but at the moment it can not be used  to build the final executable because uWebSockets.js depends on a precompiled native `.node` binary that is tightly coupled to specific Node.js versions (18, 20, 22, 23) and relies on Node-API (N-API) internals that are not fully compatible with Bun's implementation.

WE are not alone: https://github.com/oven-sh/bun/issues/9785

**Comment by puria** [comment]
Yes that's why we copy it in https://github.com/ForkbombEu/ncr/blob/main/rollup.config.js#L12-L14 

Also we do this naïve thing of rollup everything in commonjs because `pkg` do not support esm.

Anyhow... the unbelivable part is that [Bun.serve](https://bun.sh/docs/api/http) itself is based on uWebSockets

I would not touch it until it breaks... at some point ;) and maybe worth trying the native node SEA

**Comment by matteo-cristino** [comment]
It breaks here https://github.com/ForkbombEu/ncr/pull/464 😭

**Comment by matteo-cristino** [comment]
@slangroom/rdf subdeps breaks it

**Comment by puria** [comment]
ok then let's try the node23 SEA

**Comment by puria** [comment]
and take an eye on this: https://github.com/CanadaHonk/porffor
